mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -140,21 +140,27 @@ getArchAttributes(Operation *op, Type inputType) {
 }
 
 static FailureOr<rock::ConvOp>
-makeRockConv2D(ConversionPatternRewriter &rw, Operation *op, Value input,
-               Value filter, Value output, const DenseI64ArrayAttr &pad,
-               const DenseI64ArrayAttr &stride,
-               const DenseI64ArrayAttr &dilation, int64_t group) {
+makeRockConv(ConversionPatternRewriter &rw, Operation *op, Value input,
+             Value filter, Value output, const DenseI64ArrayAttr &pad,
+             const DenseI64ArrayAttr &stride, const DenseI64ArrayAttr &dilation,
+             int64_t group) {
   Location loc = op->getLoc();
 
-  SmallString<5> filterLayout("kyxc");
+  SmallString<7> filterLayout("kyxc");
   if (auto attr = op->getAttrOfType<StringAttr>("filter_layout"))
     filterLayout = attr.getValue();
-  SmallString<5> inputLayout("nhwc");
+  else if (filter.getType().template cast<ShapedType>().getRank() > 4)
```

Comment:
Could we use that `rock.actually_conv_1d` attribute here to drop those dummy dimensions?

Reply:
Yes, but I'd prefer to leave that to the hypothetical cleanup pull-request, in the interests of not breaking anything during the review close to the deadline.